pkg/runner/artillery_helper.go

```diff
@@ -0,0 +1,199 @@
+			HTTPResponses                               float64 `json:"http.responses"`
+			PluginsExpectOk                             float64 `json:"plugins.expect.ok"`
+			PluginsExpectOkStatusCode                   float64 `json:"plugins.expect.ok.statusCode"`
+			PluginsMetricsByEndpointPostRequestCodes200 float64 `json:"plugins.metrics-by-endpoint.postRequest.codes.200"`
+			VusersFailed                                float64 `json:"vusers.failed"`
+			VusersCompleted                             float64 `json:"vusers.completed"`
+			VusersCreatedByNameGetVotes                 float64 `json:"vusers.created_by_name.Get Votes"`
+			PluginsExpectOkContentType                  float64 `json:"plugins.expect.ok.contentType"`
+			PluginsMetricsByEndpointGetRequestCodes200  float64 `json:"plugins.metrics-by-endpoint.getRequest.codes.200"`
+		} `json:"counters"`
+		Histograms struct {
+			HTTPResponseTime                                HistogramMetrics `json:"http.response_time"`
+			PluginsMetricsByEndpointResponseTimePostRequest HistogramMetrics `json:"plugins.metrics-by-endpoint.response_time.postRequest"`
+			VusersSessionLength                             HistogramMetrics `json:"vusers.session_length"`
+			PluginsMetricsByEndpointResponseTimeGetRequest  HistogramMetrics `json:"plugins.metrics-by-endpoint.response_time.getRequest"`
+		} `json:"histograms"`
+		Rates struct {
+			HTTPRequestRate float64 `json:"http.request_rate"`
+		} `json:"rates"`
+		HTTPRequestRate  float64 `json:"http.request_rate"`
+		FirstCounterAt   float64 `json:"firstCounterAt"`
+		FirstHistogramAt float64 `json:"firstHistogramAt"`
+		LastCounterAt    float64 `json:"lastCounterAt"`
+		LastHistogramAt  float64 `json:"lastHistogramAt"`
+		FirstMetricAt    float64 `json:"firstMetricAt"`
+		LastMetricAt     float64 `json:"lastMetricAt"`
+		Period           string  `json:"period"`
+		Summaries        struct {
+			HTTPResponseTime                                Summary `json:"http.response_time"`
+			PluginsMetricsByEndpointResponseTimePostRequest Summary `json:"plugins.metrics-by-endpoint.response_time.postRequest"`
+			VusersSessionLength                             Summary `json:"vusers.session_length"`
+			PluginsMetricsByEndpointResponseTimeGetRequest  Summary `json:"plugins.metrics-by-endpoint.response_time.getRequest"`
+		} `json:"summaries"`
+	} `json:"intermediate"`
+}
+
+// Validate checks if Execution has valid data in context of Artillery executor
+func (r *ArtilleryRunner) Validate(execution testkube.Execution) error {
+
+	if execution.Content == nil {
+		return fmt.Errorf("can't find any content to run in execution data: %+v", execution)
+	}
+
+	if execution.Content.Repository == nil {
+		return fmt.Errorf("artillery executor handle only repository based tests, but repository is nil")
+	}
+
+	if execution.Content.Repository.Path == "" {
+		return fmt.Errorf("can't find repository path in params, repo:%+v", execution.Content.Repository)
+	}
```

Comment:
This one should be removed - just noticed that we have an issue in template (was fixing Cypress issue recently)

Reply:
But isn't this check necessary for git-based executors to check if the  test files path is not mentioned for the given git repo

Reply:
Yeah, but we also are handling `""` as root currently, I think it was some leftover in template :/

Reply:
oh alright, I have removed the check